Modules/_lzmamodule.c

```diff
@@ -355,11 +358,12 @@ lzma_filter_converter(_lzma_state *state, PyObject *spec, void *ptr)
                         "Filter specifier must have an \"id\" entry");
         return 0;
     }
-    f->id = PyLong_AsUnsignedLongLong(id_obj);
-    Py_DECREF(id_obj);
-    if (PyErr_Occurred()) {
+    lzma_vli id;
+    if (!lzma_vli_converter(id_obj, &id)) {
+        Py_DECREF(id_obj);
         return 0;
     }
```

Comment:
Py_DECREF(id_obj); is no longer needed here?

Reply:
It is needed. We own a reference to `id_obj`.

But I forget `Py_DECREF(id_obj)` in the success case.

Reply:
```suggestion
    }
    Py_DECREF(id_obj);
```